# Introduction
***

In [1]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot
import requests
import os

# Data Gathering

In [7]:
salary_df=pd.read_csv("Salary_Dataset_with_Extra_Features.csv")

In [8]:
salary_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22770 entries, 0 to 22769
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Rating             22770 non-null  float64
 1   Company Name       22770 non-null  object 
 2   Job Title          22770 non-null  object 
 3   Salary             22770 non-null  int64  
 4   Salaries Reported  22770 non-null  int64  
 5   Location           22770 non-null  object 
 6   Employment Status  22770 non-null  object 
 7   Job Roles          22770 non-null  object 
dtypes: float64(1), int64(2), object(5)
memory usage: 1.4+ MB


#### Notable Observation
- There are 22,770 rows and 8 columns
- Rename column titles to snake_case
- No empty cells at all
- Salary is integer rather than float
- Other variables are in the right format
- Minor data cleaning is needed


# Data Cleaning

In [31]:
salary_df.columns.str.lower().to_list()

['rating',
 'company name',
 'job title',
 'salary',
 'salaries reported',
 'location',
 'employment status',
 'job roles']

In [42]:
header=['rating', 'company_name','job title','salary','salaries_reported','location','employment_status','job_roles']
salary_df.iloc[0:]=salary_df
salary_df.columns=header

In [43]:
salary_df.head()

rating                      company_name          job title   salary  \
0     3.8                            Sasken  Android Developer   400000   
1     4.5  Advanced Millennium Technologies  Android Developer   400000   
2     4.0                         Unacademy  Android Developer  1000000   
3     3.8                SnapBizz Cloudtech  Android Developer   300000   
4     4.4            Appoids Tech Solutions  Android Developer   600000   

   salaries_reported   location employment_status job_roles  
0                  3  Bangalore         Full Time   Android  
1                  3  Bangalore         Full Time   Android  
2                  3  Bangalore         Full Time   Android  
3                  3  Bangalore         Full Time   Android  
4                  3  Bangalore         Full Time   Android

#### Convert salary to float

In [49]:
salary_df['salary']=salary_df["salary"].astype(float)

In [60]:
salary_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 22770 entries, 0 to 22769
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   rating             22770 non-null  float64
 1   company_name       22770 non-null  object 
 2   job title          22770 non-null  object 
 3   salary             22770 non-null  float64
 4   salaries_reported  22770 non-null  int64  
 5   location           22770 non-null  object 
 6   employment_status  22770 non-null  object 
 7   job_roles          22770 non-null  object 
dtypes: float64(2), int64(1), object(5)
memory usage: 1.4+ MB


# Data Analysis- Exploratory
***

In [61]:
df=salary_df.copy()